lib/Bailador/Request.pm

```diff
@@ -25,40 +27,61 @@ class Bailador::Request {
     }
     multi method params ($source) {
         my %ret;
+        my $source_data = '';
         given $source {
             when 'query' {
-                if $!env<QUERY_STRING> {
-                    for $.env<QUERY_STRING>.split('&') -> $p {
-                        my $pair = $p.split('=', 2);
-                        %ret{$pair[0]} = uri_unescape $pair[1];
-                    }
-                }
+                $source_data = $.env<QUERY_STRING> if $!env<QUERY_STRING>;
             }
             when 'body' {
-                if $!env<p6sgi.input> {
-                    for $.env<p6sgi.input>.decode.split('&') -> $p {
-                        my $pair = $p.split('=', 2);
-                        %ret{$pair[0]} = uri_unescape $pair[1];
-                    }
-                }
+                $source_data = $.env<p6sgi.input>.decode if $!env<p6sgi.input>;
             }
             default {
                 die "unknown source '$source'";
             }
         }
+        return %ret unless $source_data.chars;
+
+        for $source_data.split('&') -> $p {
+            my @pair = $p.split('=', 2);
+            if (%ret{@pair[0]}:exists) {
+                %ret.push((@pair[0] => |uri_unescape(@pair[1])));
+            }
+            else {
+                %ret{@pair[0]} = uri_unescape @pair[1];
+            }
+        }
 
         return %ret;
     }
 
+    method headers () {
+        return %!headers if %!headers;
+        for $.env.keys.grep(rx:i/^[HTTP||CONTENT]/) -> $key {
+            my $field = $key;
+            $field ~~ s:i/^HTTPS?_//;
```

Comment:
That's great! I was looking for a way to do it in Perl 6!

But, I can't find S// in the docs, not even in http://docs.perl6.org/language/regexes#Substitution. Here's what I tried:

```
$ perl6 -e 'my $key = q<HTTP_USER_AGENT>; my $field = $key ~~ S/HTTP//; say $field'
Potential difficulties:
    Smartmatch with S/// can never succeed because the string it returns will fail to match. You can use given instead of ~~.
    at -e:1
    ------> q<HTTP_USER_AGENT>; my $field = $key ~~ ⏏S/HTTP//; say $field
False

$ perl6 -e 'my $key = q<HTTP_USER_AGENT>; my $field = ($key ~~ S/HTTP//); say $field'
Potential difficulties:
    Smartmatch with S/// can never succeed because the string it returns will fail to match. You can use given instead of ~~.
    at -e:1
    ------> <HTTP_USER_AGENT>; my $field = ($key ~~ ⏏S/HTTP//); say $field
False
```

What am I doing wrong? The error message is not very helpful :(